### DISCUSSION TITLE
Running as systemd ... or not?

### DISCUSSION THREAD

**Discussion by johnsidoli** [abraunegg/onedrive#2080] [Q&A] [answered]
Sorry for my third question here, but I am using the client actively now and lots of little issues are popping up. This should be a simple one!

If I use the command

`systemctl`

so see all systemd stuff, onedrive is **not** there... if I use options for
`
systemctl --type=service --state=active`

it also is **not** there

if I use

```
john@lmint:~$ systemctl status onedrive@john
● onedrive@john.service - OneDrive Free Client for john
     Loaded: loaded (/lib/systemd/system/onedrive@.service; disabled; vendor preset: enabled)
     Active: inactive (dead)
       Docs: https://github.com/abraunegg/onedrive
```

it's there... but implies inactive?

But onedrive **IS** running and it **IS** working. It is uploading/downloading/logging/working. I set it up as systemd @ john and as far as I can tell it works. But all these commands say it isn't there?

How do I find it as a systemd service? I just want to stop it, resync, and restart it as per our discussion here:

https://github.com/abraunegg/onedrive/discussions/2078#discussioncomment-3332232

**Comment by abraunegg** [comment]
@johnsidoli 
> How do I find it as a systemd service?

Please review this section in the usage document: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#running-onedrive-as-a-system-service in regards to systemd services

To find all your running applications & how they are running:
```
sudo ps aufxw | grep onedrive
```

To see all the systemd services running:
```
sudo systemctl | grep onedrive
```

> But onedrive IS running and it IS working. It is uploading/downloading/logging/working. 

You could have enabled it under a different user or even have it running by mistake as 'root' .... this is especially common if you prefixed any of the commands with 'sudo' ... 

The client does not need 'sudo' to run ... it is a 'user space' application - even the systemd services - thus, lots of people make this easy mistake.

**Reply by johnsidoli** [reply]
THanks @abraunegg 

When I use `sudo systemctl | grep onedrive` I don't get anything.

Does that mean I've enabled it as root? It won't be another user, just me.

I hope it is easy to 'undo' and set as user space?

**Comment by abraunegg** [comment]
@johnsidoli 
> When I use sudo systemctl | grep onedrive I don't get anything.

Let me double check on a Ubuntu system to ensure that this will return *something*

What does `sudo ps aufxw | grep onedrive` return ?

**Reply by johnsidoli** [reply]
```
john         2359  0.0  0.0   8136   548 ?        S    20:15   0:00          |   \_ tail -n -0 -f /home/john/.config/sharepoint_projects/john.onedrive.log
john         2368  0.0  0.0   2608   596 ?        S    20:15   0:00          \_ /bin/sh /home/john/.config/onedrive/notify_OneDrive.sh
john         2372  0.0  0.0   8136   548 ?        S    20:15   0:00          |   \_ tail -n -0 -f /home/john/.config/onedrive/john.onedrive.log
john         2374  0.0  0.0   2608    88 ?        S    20:15   0:00          |   \_ /bin/sh /home/john/.config/onedrive/notify_OneDrive.sh
john         2019  1.1  0.2 480312 42360 ?        Ssl  20:15   0:57  \_ /usr/bin/onedrive --monitor
john         2021  5.6  0.2 480544 47612 ?        Ssl  20:15   4:45  \_ /usr/bin/onedrive --monitor --confdir=/home/john/.config/sharepoint_projects
john        36377  0.0  0.0   9036   716 pts/0    S+   21:40   0:00          \_ grep --color=auto onedrive
```

Returns as per that... but the other command does nothing i..e 

`sudo systemctl | grep onedrive`

**Reply by abraunegg** [reply]
Let me setup a Ubuntu system to help you further

**Comment by abraunegg** [comment]
@johnsidoli 
From your `ps` output - it appears you are starting the clients via this script rather than systemd `/home/john/.config/onedrive/notify_OneDrive.sh`

What is the contents of this file? `cat /home/john/.config/onedrive/notify_OneDrive.sh` ?

**Reply by johnsidoli** [reply]
Hi @abraunegg ... not as far as I know, the notify_OneDrive.sh is just a script I did to send new log entries to the system notifications:
```
tail -n -0 -f /home/john/.config/onedrive/john.onedrive.log | grep --line-buffered -e 'Deleting' -e 'Uploading' -e 'Moving' -e 'Downloading' -e 'ERROR' | while read line; do notify-send "Notification from OneDrive:" "$line"; done
```

**Reply by abraunegg** [reply]
Please try and use triple ` when posting output .. it is way more friendlier

**Reply by johnsidoli** [reply]
Sure! Thanks for the tip.

```tail -n -0 -f /home/john/.config/onedrive/john.onedrive.log | grep --line-buffered -e 'Deleting' -e 'Uploading' -e 'Moving' -e 'Downloading' -e 'ERROR' | while read line; do notify-send "Notification from OneDrive:" "$line"; done```

**Reply by abraunegg** [reply]
you need to also have the triple ` on a new line, rather than on the same line

**Comment by abraunegg** [comment, marked as answer]
@johnsidoli 
Suggestions for trying to find these systemd services:

**systemd running as a root user**
```
sudo systemctl --user list-units --type=service | grep onedrive
```

**systemd running as a root system**
```
sudo systemctl list-units --type=service | grep onedrive
```

**systemd running as your logged in user**
```
systemctl --user list-units --type=service | grep onedrive
```

**systemd running as your logged in user, but running as root (maybe)**
```
systemctl list-units --type=service | grep onedrive
```

**Comment by abraunegg** [comment]
@johnsidoli 
Additionally, https://github.com/abraunegg/onedrive/blob/master/docs/SharePoint-Shared-Libraries.md has been updated to clarify many items.

Please review and work through this updated document to configure your SharePoint Libraries.

**Comment by johnsidoli** [comment]
Thanks @abraunegg - the "systemd running as your logged in user" command finally found it!

```john@mint:~$ systemctl --user list-units --type=service | grep onedrive```
```onedrive.service                      loaded active running OneDrive Free Client```

I looked through the sharepoint doc, but looks the same as when I setup to me!

**Comment by abraunegg** [comment]
@johnsidoli 
Great - so the original question:

> How do I find it as a systemd service? I just want to stop it, resync, and restart it as per our discussion here

is now solved - you now know the systemd service that is running .. the remaining issue however is - if this *default* service running a SharePoint Library .. if so - it really should be it's own configuration rather than using the *default*
